app/controllers/SeriesDetailsController.scala

```diff
@@ -27,35 +29,49 @@ class SeriesDetailsController @Inject()(val controllerComponents: SecurityCompon
-    seriesService.getSeriesForUser(request.token)
-      .map({series =>
-        val seriesFormData = series.map(s => (s.seriesid.toString, s.code))
-        status(views.html.standard.seriesDetails(seriesFormData, form, request.token.name))
-      })
+  private def getSeriesDetails(consignmentId: UUID, request: Request[AnyContent], status: Status, form: Form[SelectedSeriesData])
+                              (implicit requestHeader: RequestHeader) = {
+    consignmentStatusService.consignmentStatusSeries(consignmentId, request.token.bearerAccessToken).flatMap {
+      consignmentStatus =>
+        val seriesStatus = consignmentStatus.flatMap(_.currentStatus.series)
+        seriesStatus match {
+          case Some("Completed") =>
+            val seriesFormData = List((consignmentStatus.flatMap(_.series.map(_.seriesid.toString)).get, consignmentStatus.flatMap(_.series.map(_.code)).get))
+            Future(Ok(views.html.standard.seriesDetailsAlreadyConfirmed(consignmentId, seriesFormData, selectedSeriesForm, request.token.name)).uncache())
+          case _ =>
+            seriesService.getSeriesForUser(request.token)
+              .map { series =>
+                val seriesFormData = series.map(s => (s.seriesid.toString, s.code))
+                status(views.html.standard.seriesDetails(consignmentId, seriesFormData, form, request.token.name)).uncache()
+              }
+        }
+    }
   }
 
-  def seriesDetails(): Action[AnyContent] = standardUserAction { implicit request: Request[AnyContent] =>
-    getSeriesDetails(request, Ok, selectedSeriesForm)
+  def seriesDetails(consignmentId: UUID): Action[AnyContent] = standardUserAction { implicit request: Request[AnyContent] =>
+    getSeriesDetails(consignmentId, request, Ok, selectedSeriesForm)
   }
 
-  def seriesSubmit(): Action[AnyContent] =  standardUserAction { implicit request: Request[AnyContent] =>
+  def seriesSubmit(consignmentId: UUID): Action[AnyContent] =  standardUserAction { implicit request: Request[AnyContent] =>
     val formValidationResult: Form[SelectedSeriesData] = selectedSeriesForm.bindFromRequest()
 
     val errorFunction: Form[SelectedSeriesData] => Future[Result] = { formWithErrors: Form[SelectedSeriesData] =>
-      getSeriesDetails(request, BadRequest, formWithErrors)
+      getSeriesDetails(consignmentId, request, BadRequest, formWithErrors)
     }
 
     val successFunction: SelectedSeriesData => Future[Result] = { formData: SelectedSeriesData =>
-      consignmentService
-        .createConsignment(Some(UUID.fromString(formData.seriesId)), request.token)
-        .map { consignment =>
-          if (request.token.isJudgmentUser) {
-            Redirect(routes.BeforeUploadingController.beforeUploading(consignment.consignmentid.get))
-          } else {
-            Redirect(routes.TransferAgreementPrivateBetaController.transferAgreement(consignment.consignmentid.get))
-          }
+      if (request.token.isJudgmentUser) {
```

Comment:
This bit isn't needed. The action prevents any judgment users from getting to this page so we don't need to check for it here. You can just have lines 66 to 73 instead of the if/else